File: src/commands/ExecuteScriptCommand.java
package commands;

import client.Application;
import commands.core.ClientSideCommand;
import lib.ExecutionStatus;
import lib.functions.CallbackFunction;

import java.io.FileNotFoundException;
import java.io.FileReader;

public class ExecuteScriptCommand extends ClientSideCommand {
    public ExecuteScriptCommand(CallbackFunction callbackFunction, Application application) {
        super(callbackFunction, application);
    }

    @Override
    protected void executeImplementation(String[] args, CallbackFunction callback) throws FileNotFoundException {
        String filename = args[0];
        FileReader fileReader = new FileReader(filename);

        application.getBufferedReaderWithQueueOfStreams().pushStream(fileReader);
        callback.invoke(ExecutionStatus.SUCCESS, null, filename);
    }
}

File: src/commands/client_side/ExecuteScriptCommand.kt
package commands.client_side;

import application.Application;
import commands.client_side.core.ClientSideCommand;
import lib.ExecutionStatus;
import commands.client_side.core.ClientCallbackFunction;

import java.io.FileNotFoundException;
import java.io.FileReader;

public class ExecuteScriptCommand extends ClientSideCommand {
    public ExecuteScriptCommand(ClientCallbackFunction clientCallbackFunction, Application application) {
        super(clientCallbackFunction, application);
    }

    @Override
    protected void executeImplementation(String[] args, ClientCallbackFunction callback) throws FileNotFoundException {
        String filename = args[0];

        try (FileReader fileReader = new FileReader(filename)) {
            application.getBufferedReaderWithQueueOfStreams().pushStream(fileReader);
            callback.invoke(ExecutionStatus.SUCCESS, null, filename);
        } catch (Exception e) {
            callback.invoke(ExecutionStatus.FAILURE, e, filename);
        }
    }
}
